src/state.rs

```diff
@@ -179,6 +180,10 @@ impl State {
         (Err(encrypted_container::Error::FailedDecryption), None)
     }
 
+    pub fn file_exists<T: FilesystemClient>(&mut self, trussed: &mut T, filename: PathBuf) -> bool {
+        try_syscall!(trussed.read_file(self.location, filename)).is_ok()
```

Comment:
The [`entry_metadata`](https://github.com/trussed-dev/trussed/blob/main/src/client.rs#L649C12-L649C12) syscall was made for this use case.

Reply:
Thanks! Will change to it. 
Note that the name is not really intuitive for this use case though. Does it not call for a proper alias?

Reply:
You mean check_exists?
Entry metadata does more than just checking existence.

Reply:
`check_exists` is not a good name. There is `exists` in the API already, and it refers to a key, not a file.
`file_exists` is the intuitive one as far as I see.

Reply:
Moved check to entry_metadata. Resolving.